src/ui/public/field_editor/field_editor.js

```diff
@@ -61,8 +61,12 @@ uiModules
         const fields = indexPattern.fields;
         const field = self.field.toActualField();
 
-        fields.remove({ name: field.name });
-        fields.push(field);
+        const index = fields.findIndex(f => f.name === field.name);
+        if (index > -1) {
+          fields.splice(index, 1, field);
```

Comment:
Good call, didn't think about that. I'm generally hesitant to use assignment with arrays, but since the index is coming from `.findIndex()` it should be good, and I agree that it will be easier to read.